inbm/cloudadapter-agent/tests/unit/agent/test_publisher.py

```diff
@@ -330,3 +448,75 @@ def test_publish_config_without_arguments_fail(self):
         except ValueError:
             failed = True
         assert failed
+
+    def test_query_all(self):
+        arguments = self.QUERY_ARGUMENTS_ALL
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_ALL)
+
+    def test_query_hw(self):
+        arguments = self.QUERY_ARGUMENTS_HW
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_HW)
+
+    def test_query_fw(self):
+        arguments = self.QUERY_ARGUMENTS_FW
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_FW)
+
+    def test_query_os(self):
+        arguments = self.QUERY_ARGUMENTS_OS
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_OS)
+
+    def test_query_security(self):
+        arguments = self.QUERY_ARGUMENTS_SECURITY
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_SECURITY)
+
+    def test_query_swbom(self):
+        arguments = self.QUERY_ARGUMENTS_SWBOM
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_SWBOM)
+
+    def test_query_guid(self):
+        arguments = self.QUERY_ARGUMENTS_GUID
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_GUID)
+
+    def test_query_status(self):
+        arguments = self.QUERY_ARGUMENTS_STATUS
+
+        message = self.publisher.publish_query(**arguments)
+
+        assert message == MESSAGE.QUERY
+        mocked = self.MockBroker.return_value
+        mocked.publish_install.assert_called_once_with(self.QUERY_STATUS)
```

Comment:
We will need a negative test in the case an invalid option is given.  Just as he has above for  test_publish_config_with_invalid_cmd_fail(self):